src/components/detailsProduct/detailsProdact.ts

```diff
@@ -0,0 +1,167 @@
+import { IFiltersElems, IProducts } from '../../models';
+import { createElem } from '../../utilities';
+import ModalPayment from '../modalPayment/modalPayment';
+import './detailsProduct.scss';
+
+export default class Details {
+  btnAdd: HTMLButtonElement | null = null;
+  imagesElems: IFiltersElems = {};
+
+  renderDetails(id?: number): void {
+    if (!id) {
+      this.renderPage404();
+      return;
+    }
+    const productObj = this.getProductById(id);
+    if (!productObj) {
+      this.renderPage404();
+      return;
+    }
+
+    const main = document.querySelector('.main') as HTMLElement;
+    if (!main) return;
+    main.innerHTML = this.drawDetailsProduct(productObj);
+    this.drawImagesDetailsProduct(productObj, main);
+
+    const hasBasketProduct = window.app.dataBase.checkProductInBasket(productObj);
+    this.btnAdd = main.querySelector('.add-to-cart-btn') as HTMLButtonElement;
+    if (hasBasketProduct) {
+      this.btnAdd.classList.add('added');
+    }
+    this.btnAdd.textContent = hasBasketProduct ? 'Drop from Cart' : 'Add to Cart';
```

Comment:
DRY - dont repeat yourself. If you use `Drop from Cart` more than 1 time, move it in constants and import